aggregate/billing_functions/utils.py

```diff
@@ -0,0 +1,742 @@
+    when we find a batch that started before the date.
+    """
+    batches = []
+    last_batch_id = None
+    n_requests = 0
+    skipped = 0
+
+    logger.info(f'Getting batches for range: [{start}, {end}]')
+
+    while True:
+
+        n_requests += 1
+        jresponse = await get_batches(
+            billing_project=billing_project, last_batch_id=last_batch_id, token=token
+        )
+
+        if 'last_batch_id' in jresponse and jresponse['last_batch_id'] == last_batch_id:
+            raise ValueError(
+                f'Something weird is happening with last_batch_job: {last_batch_id}'
+            )
+        last_batch_id = jresponse.get('last_batch_id')
+        if not jresponse.get('batches'):
+            return batches
+        for b in jresponse['batches']:
+            if not b['time_completed'] or not b['complete']:
+                skipped += 1
+                continue
+
+            time_completed = parse_hail_time(b['time_completed'])
+            in_date_range = start <= time_completed < end
+
+            if time_completed < start:
+                logger.info(
+                    f'{billing_project} :: Got {len(batches)} batches '
+                    f'in {n_requests} requests, skipping {skipped}'
+                )
+                return batches
+            if in_date_range:
+                batches.append(b)
+            else:
+                skipped += 1
+
+
+async def get_jobs_for_batch(batch_id, token: str) -> List[str]:
+    """
+    For a single batch, fill in the 'jobs' field.
+    """
+    jobs = []
+    last_job_id = None
+    end = False
+    iterations = 0
+
+    async with aiohttp.ClientSession() as session:
+        while not end:
+            iterations += 1
+
+            if iterations > 1 and iterations % 5 == 0:
+                logger.info(f'On {iterations} iteration to load jobs for {batch_id}')
+
+            q = '?limit=9999'
+            if last_job_id:
+                q += f'&last_job_id={last_job_id}'
+            url = HAIL_JOBS_API.format(batch_id=batch_id) + q
+
+            jresponse = await async_retry_transient_get_json_request(
+                url,
+                aiohttp.ClientError,
+                session=session,
+                headers={'Authorization': 'Bearer ' + token},
+            )
+            new_last_job_id = jresponse.get('last_job_id')
+            if new_last_job_id is None:
+                end = True
+            elif last_job_id and new_last_job_id <= last_job_id:
+                raise ValueError('Something fishy with last job id')
+            last_job_id = new_last_job_id
+            jobs.extend(jresponse['jobs'])
+
+    return jobs
+
+
+RE_matcher = re.compile(r'-\d+$')
+
+
+def billing_row_to_topic(row, dataset_to_gcp_map) -> Optional[str]:
+    """Convert a billing row to a dataset"""
+    project_id = row['project']['id']
+    topic = dataset_to_gcp_map.get(project_id, project_id)
+    if not topic:
+        return 'admin'
```

Comment:
Define this as constant somewhere and add a comment

Reply:
Added constant to top, set as DEFAULT_TOPIC